### PR TITLE
Pattern name bug fix

### PR DESCRIPTION
**Pull request by br648**
### Checklist

- [x] Appropriate branch selected _(all PRs must first be merged to `dev` before they can be merged to `master`)_
- [x] Any modified or new methods or classes have helpful JavaDoc and code is thoroughly commented
- [x] The description lists all applicable issues this PR seeks to resolve
- [x] The description lists any configuration setting(s) that differ from the default settings
- [x] All tests and CI builds passing

### Description

Update to use the pattern ids and names provided with a feed to the patterns generated in the pattern finder. The pattern ids are then used in the pattern builder to generate pattern stops with the correctly related pattern ids.

Related DT PR: https://github.com/ibi-group/datatools-server/pull/586

### CODE REVIEW

File: src/main/java/com/conveyal/gtfs/validator/PatternFinderValidator.java [hunk 1]
```diff
@@ -72,19 +73,18 @@ public void validateTrip (Trip trip, Route route, List<StopTime> stopTimes, List
...
         LOG.info("Finding patterns...");
         Map<String, Stop> stopById = new HashMap<>();
         for (Stop stop : feed.stops) {
             stopById.put(stop.stop_id, stop);
         }
         // Although patterns may have already been loaded from file, the trip patterns are still required.
-        Map<TripPatternKey, Pattern> patterns = patternFinder.createPatternObjects(stopById, errorStorage);
-        patternBuilder.create(patterns, patternIds);
+        Map<TripPatternKey, Pattern> patterns = patternFinder.createPatternObjects(stopById, patternsFromFeed, errorStorage);
+        patternBuilder.create(patterns, !patternsFromFeed.isEmpty());
```
Comment (binh-dam-ibigroup) [inline src/main/java/com/conveyal/gtfs/validator/PatternFinderValidator.java]: Replace `!patternsFromFeed.isEmpty()` with `patternFinder.canUsePatternsFromFeed`. As written, I don't see any pattern renaming taking place either after importing a feed into a blank editor where ` canUsePatternsFromFeed` is false, or when viewing the "Patterns" tab in the feed version summary page.

**Review by binh-dam-ibigroup — CHANGES_REQUESTED** [review]
Fix the indicated gap in the logic that prevents the pattern renaming to take place and we'll be good to go.

File: src/main/java/com/conveyal/gtfs/validator/PatternFinderValidator.java (hunk 1 above)
Comment (br648) [reply src/main/java/com/conveyal/gtfs/validator/PatternFinderValidator.java]: Fixed in: https://github.com/ibi-group/gtfs-lib/pull/6/commits/287badffe2860e94a320bcda62fee33213069541

File: src/main/java/com/conveyal/gtfs/PatternBuilder.java [hunk 2]
```diff
@@ -40,7 +39,7 @@ public PatternBuilder(Feed feed) throws SQLException {
         connection = feed.getConnection();
     }
 
-    public void create(Map<TripPatternKey, Pattern> patterns, Set<String> patternIdsLoadedFromFile) {
+    public void create(Map<TripPatternKey, Pattern> patterns,  boolean usePatternsFromFeed) {
```
Comment (binh-dam-ibigroup) [inline src/main/java/com/conveyal/gtfs/PatternBuilder.java]: Delete extra space before `boolean`.

**Review by binh-dam-ibigroup — APPROVED** [review]
All good now! Please fix one tiny formatting glitch.

**Review by miles-grant-ibigroup — APPROVED** [review]
Great approach, thanks!

**Pull request merged**